src/components/layout/GradientBottomActions.tsx

```diff
@@ -1,18 +1,23 @@
 import * as React from "react";
-import { View, StyleSheet, ViewStyle, StyleProp } from "react-native";
+import { StyleProp, StyleSheet, View, ViewStyle } from "react-native";
 import LinearGradient from "react-native-linear-gradient";
 import Animated from "react-native-reanimated";
+import { IOColors, IOSpacer, IOVisualCostants, hexToRgba } from "../../core";
 import { WithTestID } from "../../utils/types";
-import { IOColors, hexToRgba, IOSpacer, IOVisualCostants } from "../../core";
+import {
+  ButtonLink,
+  ButtonLinkProps,
+  ButtonSolid,
+  ButtonSolidProps
+} from "../buttons";
 import { VSpacer } from "../spacer";
 
 export type GradientBottomActions = WithTestID<{
   transitionAnimStyle: Animated.AnimateStyle<StyleProp<ViewStyle>>;
   dimensions: GradientBottomActionsDimensions;
-  // Accepted components: ButtonSolid, ButtonLink
-  // Don't use any components other than this, please.
-  primaryAction?: React.ReactNode;
-  secondaryAction?: React.ReactNode;
+  // Accepted components: ButtonSolid for the primaryAction, ButtonLink for the secondaryAction
+  primaryActionProps?: ButtonSolidProps;
```

Comment:
The primary button should be always rendered with `fullWidth` set to `true`, so we can exclude that prop to avoid redundant (and ineffective) props:
```suggestion
  primaryActionProps?: Omit<ButtonSolidProps, "fullWidth">;
```